cmd/legacy_param_mapper_test.go

```diff
@@ -504,3 +504,53 @@ func TestInvalidClientProtocol(t *testing.T) {
 
 	assert.NotNil(t, err)
 }
+
+func TestEnableEmptyManagedFoldersVetting(t *testing.T) {
+	testcases := []struct {
+		name                              string
+		enableHns                         bool
+		enableEmptyManagedFolders         bool
+		expectedEnableEmptyManagedFolders bool
+	}{
+		{
+			name:                              "both enable-hns and enable-empty-managed-folders set to true",
+			enableHns:                         true,
+			enableEmptyManagedFolders:         true,
+			expectedEnableEmptyManagedFolders: true,
+		},
+		{
+			name:                              "enable-hns set to true",
+			enableHns:                         true,
+			enableEmptyManagedFolders:         false,
+			expectedEnableEmptyManagedFolders: true,
+		},
+		{
+			name:                              "enable-hns not true but enable-empty-managed-folders set to true",
+			enableHns:                         false,
+			enableEmptyManagedFolders:         true,
+			expectedEnableEmptyManagedFolders: true,
+		},
+		{
+			name:                              "both enable-hns and enable-empty-managed-folders not true",
+			enableHns:                         false,
+			enableEmptyManagedFolders:         false,
+			expectedEnableEmptyManagedFolders: false,
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			flags := &flagStorage{
+				ClientProtocol: mountpkg.ClientProtocol("http1"),
+			}
+			c := config.NewMountConfig()
+			c.EnableHNS = tc.enableHns
+			c.ListConfig.EnableEmptyManagedFolders = tc.enableEmptyManagedFolders
+
+			resolvedConfig, err := PopulateNewConfigFromLegacyFlagsAndConfig(&mockCLIContext{}, flags, c)
+
+			if assert.NoError(t, err) {
```

Comment:
When would `assert.NoError(t, err)` be false ?

Reply:
> When error is not nil.

No, I understood that, but where will the error in it come from? If it's from the test case, then we should have had a `break` on the previous test-case failure rather than calling the function `PopulateNewConfigFromLegacyFlagsAndConfig` in this iteration, but not using/testing its value. This seems redundant.

Reply:
I get it. My point is that we're skipping the test for `resolvedConfig.List.EnableEmptyManagedFolders` if `PopulateNewConfigFromLegacyFlagsAndConfig(&mockCLIContext{}, flags, c)` failed. This sort of reduces the effectiveness of this test.

Is it possible for `PopulateNewConfigFromLegacyFlagsAndConfig(&mockCLIContext{}, flags, c)` in this scenario to fail?

Reply:
If `PopulateNewConfigFromLegacyFlagsAndConfig` returns an error, the test fails. Also, if error is returned, the value of resolvedConfig could be anything - there are no expectations on what its value is. The only expectation is that it should not be used. So, there is no point asserting on the value.

> Is it possible for PopulateNewConfigFromLegacyFlagsAndConfig(&mockCLIContext{}, flags, c) in this scenario to fail?

I guess I'm not getting your concern. If the call fails, the test fails (assert.NoError(...) also causes the test to fail) and if it doesn't we assert on the value. So, I guess we are covered regardless of whether the call fails.